var/spack/repos/builtin/packages/arborx/package.py

```diff
@@ -88,18 +88,16 @@ def cached_tests_work_dir(self):
     def build_tests(self):
         """Build the stand-alone/smoke test."""
 
-        # We don't need to append the path to Kokkos to CMAKE_PREFIX_PATH since
-        # a hint is already hardcoded inside the CMake ArborX configuration.
-        # Omitting it here allows us to avoid to distinguish between Kokkos
-        # being installed as a standalone or as part of Trilinos.
         arborx_dir = self.spec['arborx'].prefix
         cmake_prefix_path = "-DCMAKE_PREFIX_PATH={0}".format(arborx_dir)
         if '+mpi' in self.spec:
             cmake_prefix_path += ";{0}".format(self.spec['mpi'].prefix)
 
         cmake_args = [".",
                       cmake_prefix_path,
-                      "-DCMAKE_CXX_COMPILER={0}".format(self.compiler.cxx)]
+                      "-DCMAKE_CXX_COMPILER={0}".format(self.compiler.cxx),
```

Comment:
I wonder if this actually works for `cuda`. When we build ArborX, we do a special thing for `cuda`, taking the compiler for Kokkos. Wonder if we should do the same here. Outside of the scope of the PR, though.

Reply:
With recent Kokkos versions, we should actually not need this workaround to set the compiler for `CUDA` since we are using `nvcc_wrapper` internally anyway such that only the host compilers need to be compatible. On the other hand, it's normally a good idea to use the same compiler as `Kokkos` does anyway to avoid corner cases. I'm not quite sure if we want to change anything then.

Reply:
What version of Kokkos started using `nvcc_wrapper` internally? Pretty sure 3.1 does not.

Reply:
Should be 3.3, see https://github.com/kokkos/kokkos/commit/0ff5a05b554a3c5d0f893dc602bda4343fb67b77.